src/services/drive/delete-file.ts

```diff
@@ -106,3 +126,35 @@ export async function deleteObjectStorageFile(key: string) {
 		Key: key
 	}).promise();
 }
+
+async function findRelatedNotes(fileId: string) {
+	// NOTE: When running raw query, TypeORM converts field name to lowercase. Wrap in quotes to prevent conversion.
+	const relatedNotes = await Notes.createQueryBuilder('note').where(':id = ANY("fileIds")', { id: fileId }).getMany();
+	for (const relatedNote of relatedNotes) {
+		const user = await Users.findOne({ id: relatedNote.userId });
+		if (user)
+			relatedNote.user = user;
+	}
+	return relatedNotes;
+}
+
+async function findCascadingNotes(note: Note) {
+	const cascadingNotes: Note[] = [];
+
+	const recursive = async (noteId: string) => {
+		const replies = await Notes.find({ replyId: noteId });
```

Comment:
It would be better to get `user` with JOIN.
```suggestion
		const query = Notes.createQueryBuilder('note')
			.where('note.replyId = :noteId', { noteId })
			.leftJoinAndSelect('note.user', 'user');

		const replies = await query.getMany();
```